lib/realtime/block_waiver.ex

```diff
@@ -34,7 +34,7 @@ defmodule Realtime.BlockWaiver do
     block
     |> Enum.flat_map(&trip_stop_time_waivers/1)
     |> group_consecutive_sequences()
-    |> Enum.map(&from_trip_stop_time_waivers/1)
+    |> Enum.map(&from_trip_stop_time_waivers(&1, date_for_block(block)))
```

Comment:
Yeah, this will correctly handle the day rollover 👍 

I'm a little worried about performance. For every block waiver, it will calculate all the active blocks, and then search through all of them. We could probably reduce some of that work with [Refactor: Make the BlockWaivers structs on write instead of read](https://app.asana.com/0/1148853526253426/1163344518291749) by fetching the active blocks only once and using it for all block waivers. I think it's probably fine to wait for that unless you've noticed it being a problem.